### PR TITLE
Show thoughts that are under meta cursor

### LINKED ISSUE #1015: Hidden thoughts should always be shown under meta attributes
## Steps to Reproduce

1. Create thought `=a`
2. Create subthought `=b` within `=a`
3. Move cursor to `=a`

i.e.

- `=a` [cursor]
  - `=b`

## Current Behavior

`=b` is hidden and `=a` appears to be a leaf.

## Expected Behavior

Do not hide metaprogramming descendants when the cursor is on a metaprogramming attribute:

- `=a` [cursor]
  - `=b` [shown]
  
The current behavior should still be used if the cursor is not on a metaprogramming attribute:

- `a` [cursor]
  - `=b` [hidden]

### PR DESCRIPTION
**Pull request by shresthabijay**
fixes #1015

# Changes

- Update `isChildVisibleWithCursorCheck` to allow showing thoughts that are under meta cursor.
- Update `nextThought` to use `firstVisibleChildWithCursorCheck`.
- Add tests.

### CODE REVIEW

File: src/components/__tests__/Subthoughts.ts [hunk 1]
```diff
@@ -395,6 +395,111 @@ describe('hidden thoughts', () => {
...
+
+    // meta2 should should not be visible when it doesn't lie in cursor path or is not descendant of meta cursor
+    expect(metaThought2()).toHaveLength(0)
+
+    store.dispatch(setCursorFirstMatchActionCreator(['a', '=meta1']))
+
+    wrapper.update()
+
+    // meta2 should should be visible when it is descendant of the meta cursor
+    expect(metaThought2()).toHaveLength(0)
```
Comment (raineorshine) [inline src/components/__tests__/Subthoughts.ts:427]: This is testing if meta2 is visible, and yet it asserts `.toHaveLength(0)`. Don't we want it to have length 1?

File: src/components/__tests__/Subthoughts.ts [hunk 2]
```diff
@@ -395,6 +395,111 @@ describe('hidden thoughts', () => {
...
+      - d (~)
+       - b.d
+       - a.d
+        - e
+         - =meta1 (meta cursor)
+          - =meta2 (expecting =meta2 to be visible too)
+    */
+
+    // meta2 should should be visible when it is descendant of the meta cursor
+    expect(metaThought2()).toHaveLength(0)
```
Comment (raineorshine) [inline src/components/__tests__/Subthoughts.ts:501]: same

**Comment by shresthabijay** [comment]
@raineorshine  I have fixed the tests. Please review it!

**Comment by raineorshine** [comment]
Great, thanks!

**Pull request merged**